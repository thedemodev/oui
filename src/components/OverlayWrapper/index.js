import React from 'react';
import PropTypes from 'prop-types';
import Tether from 'tether';

/**
 * Intelligently position elements on a page.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

class OverlayWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isOverlayOpen: false };
    this.onChildClick = this.onChildClick.bind(this);
    this.onChildMouseOver = this.onChildMouseOver.bind(this);
    this.onChildMouseOut = this.onChildMouseOut.bind(this);
    this.disableTether = this.disableTether.bind(this);
  }

  getChildContext() {
    return {
      hideOverlay: this.disableTether,
    };
  }

  componentDidMount() {
    let options = {
      element: this._overlayEl,
      target: this._activatorEl.firstChild,
      attachment: `${this.props.verticalAttachment} ${this.props.horizontalAttachment}`,
      constraints: [{
        to: 'window',
        attachment: this.props.attachmentConstraint,
        pin: this.props.isConstrainedToScreen,
      }],
    };

    if (this.props.verticalTargetAttachment && this.props.horizontalTargetAttachment) {
      options.targetAttachment = `${this.props.verticalTargetAttachment} ${this.props.horizontalTargetAttachment}`;
    }

    this.initTether = () => new Tether(options);
  }

  enableTether() {
    // If tether hasn't yet been created, do it now.
    if (!this._tether) {
      this._tether = this.initTether();
    }

    this.setState({ 'isOverlayOpen': true });
    this._tether.enable();

    if (typeof this.props.onShow === 'function') {
      this.props.onShow();
    }

    // Reposition once the overlay is visible because Tether can't
    // properly calculate the positioning when the overlay is not
    // displayed.
    setTimeout(() => {
      this._tether.position();
    });

    // These functions must be stored so that it can be still be removed even
    // though `bind` was used: http://stackoverflow.com/a/22870717/316602
    this._bodyClickListener = this.props.shouldHideOnClick && this.isClickWithinOverlayOrChildren.bind(this);
    this._documentEscapeListener = this.onEscapeKey.bind(this);

    if (this._bodyClickListener) {
      document.addEventListener('click', this._bodyClickListener);
    }

    document.addEventListener('keyup', this._documentEscapeListener);
  }

  disableTether() {
    // If tether hasn't been created, return early
    if (!this._tether) {
      return;
    }

    let shouldHide = true;
    if (typeof this.props.onHide === 'function') {
      shouldHide = this.props.onHide();
    }

    if (shouldHide === false) {
      // Exit out of this function before hiding the `overlay` if the supplied
      // `onHide` function returns false. This is useful for fine-tuned control
      // of when the `overlay` should close.
      return;
    }

    this.setState({ 'isOverlayOpen': false });
    this._tether.disable();
    this.removeBodyEventListener();
  }

  removeBodyEventListener() {
    if (this._bodyClickListener) {
      document.removeEventListener('click', this._bodyClickListener);
    }
    if (this._documentEscapeListener) {
      document.removeEventListener('keyup', this._documentEscapeListener);
    }
  }

  /**
    * Disable Tether if a user clicks outside of the `overlay` and `children`.
    * @param {Object} event - Click event
   */
  isClickWithinOverlayOrChildren(event) {
    let target = event.target;
    let shouldCloseOverlay = true;

    while (target.parentNode !== null && shouldCloseOverlay) {
      if (target === this._overlayEl || target === this._activatorEl) {
        shouldCloseOverlay = false;
      }
      target = target.parentNode;
    }

    if (shouldCloseOverlay) {
      this.disableTether();
    }
  }

  componentWillUnmount() {
    this.removeBodyEventListener();
    // If tether hasn't been created, return early
    if (!this._tether) {
      return;
    }
    this._tether.destroy();
    if (this._tether.element && this._tether.element.remove) {
      this._tether.element.remove();
    }
  }

  onChildClick(event, child) {
    // Toggle the `overlay` by enabling or disabling Tether.
    if (this.state.isOverlayOpen) {
      // Disable Tether when not visible for performance reasons.
      this.disableTether();
    } else {
      // Enable Tether when visible.
      this.enableTether();
    }

    // Run the `children`'s `onClick` if it exists.
    if (child.props.onClick) {
      child.props.onClick(event);
    }
  }

  onChildMouseOver(event, child) {
    // Enable Tether when visible.
    this.enableTether();

    // Run the `children`'s `onMouseOver` if it exists.
    if (child.props.onMouseOver) {
      child.props.onMouseOver(event);
    }
  }

  onChildMouseOut(event, child) {
    // Disable Tether when not visible for performance reasons.
    this.disableTether();

    // Run the `children`'s `onMouseOut` if it exists.
    if (child.props.onMouseOut) {
      child.props.onMouseOut(event);
    }
  }

  onEscapeKey(event) {
    // Escape key
    if (event.keyCode === 27) {
      this.disableTether();
    }
  }

  render() {
    const {
      behavior,
      children,
      overlay,
      testSection,
    } = this.props;
    let eventHandlerProps;
    let child = React.Children.only(children);

    switch (behavior) {
      case 'click':
        eventHandlerProps = {
          onClick: (event) => this.onChildClick.call(null, event, child),
        };
        break;
      case 'hover':
        eventHandlerProps = {
          onMouseOver: (event) => this.onChildMouseOver.call(null, event, child),
          onMouseOut: (event) => this.onChildMouseOut.call(null, event, child),
        };
        break;
      default:
    }

    const block = { display: 'inline-block' };
    const overlayState = this.state.isOverlayOpen ? { display: 'block' } : { display: 'none' };

    return (
      /* eslint-disable react/jsx-no-bind */
      <div
        data-oui-component={ true }
        data-test-section={ testSection }
        style={ block }>
        <div
          className='click-area'
          style={ block }
          ref={ (ref) => { this._activatorEl = ref; } }
          { ...eventHandlerProps }>
          { child }
        </div>
        <div
          ref={ (ref) => { this._overlayEl = ref; } }
          style={ overlayState }>
          { overlay }
        </div>
      </div>
      /* eslint-enable */
    );
  }
}

OverlayWrapper.propTypes = {
  /** Determines how the attachment is constrained to the target */
  attachmentConstraint: PropTypes.oneOf(['both', 'element', 'none', 'target', 'together']),
  /** Event to listen to and open the overlay */
  behavior: PropTypes.oneOf(['click', 'hover']),
  /** Element that the `overlay` should attach to */
  children: PropTypes.node.isRequired,
  /** Side of the `overlay` that should attach to the `children` */
  horizontalAttachment: PropTypes.oneOf(['left', 'center', 'right']),
  /** Side of `children` that should attach to the `overlay` */
  horizontalTargetAttachment: PropTypes.oneOf(['left', 'center', 'right']),
  /** Attach `overlay` to an edge of the screen if it is going to move off */
  isConstrainedToScreen: PropTypes.bool,
  /**
   * Function that runs when the `overlay` is hidden. Return `false` to prevent
   * the `overlay` from closing.
   */
  onHide: PropTypes.func,
  /** Function that runs when the `overlay` is shown */
  onShow: PropTypes.func,
  /** The element that is attached to the children */
  overlay: PropTypes.node.isRequired,
  /** Should the `overlay` close when clicking outside of it */
  shouldHideOnClick: PropTypes.bool,
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** Vertical edge of the `overlay` that should touch the `children` */
  verticalAttachment: PropTypes.oneOf(['top', 'middle', 'bottom']),
  /** Vertical edge of the `children` that should touch the `overlay` */
  verticalTargetAttachment: PropTypes.oneOf(['top', 'middle', 'bottom']),
};

OverlayWrapper.defaultProps = {
  attachmentConstraint: 'together',
  behavior: 'click',
  horizontalAttachment: 'center',
  isConstrainedToScreen: false,
  onHide: () => {},
  shouldHideOnClick: true,
  verticalAttachment: 'top',
};

OverlayWrapper.childContextTypes = {
  hideOverlay: PropTypes.func,
};

export default OverlayWrapper;
