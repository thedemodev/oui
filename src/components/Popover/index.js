import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const DEFAULT_PADDING_PROP = 'default';
const HARD_PADDING_PROP = 'hard';
const SOFT_DOUBLE_PROP = 'soft-double';
const SOFT_HALF_PADDING_PROP = 'soft-half';

/**
 * Simple component often used to display supplemental information to contents
 * on a page.
 *
 * See `OverlayWrapper` component to position a popover on a page.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */
const Popover = ({
  children,
  padding,
  testSection,
  title,
}) => {
  const popOverClasses = classNames({
    'oui-pop--over': true,
  });

  let popOverContentClasses = classNames({
    'oui-pop--over__content': true,
  });

  const shouldAddPaddingClass = padding !== DEFAULT_PADDING_PROP && [
    HARD_PADDING_PROP,
    SOFT_DOUBLE_PROP,
    SOFT_HALF_PADDING_PROP,
  ].includes(padding);

  if (shouldAddPaddingClass) {
    popOverContentClasses = classNames({
      'oui-pop--over__content': true,
      [padding]: true,
    });
  }

  return (
    <div
      data-oui-component={ true }
      className={ popOverClasses }
      style={{ display: 'block', opacity: 1, position: 'initial' }}
      data-test-section={ testSection }>
      <div className={ popOverContentClasses }>
        { title && (
          <div className="oui-pop--over__title">{ title }</div>
        ) }
        { children }
      </div>
    </div>
  );
};

Popover.propTypes = {
  /** Content that appears within the popover body */
  children: PropTypes.node.isRequired,
  /** Padding class or 'default' padding for the popover container */
  padding: PropTypes.oneOf([DEFAULT_PADDING_PROP, HARD_PADDING_PROP, SOFT_DOUBLE_PROP, SOFT_HALF_PADDING_PROP]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
  /** Text describing the popover contents */
  title: PropTypes.string,
};

Popover.defaultProps = {
  padding: 'default',
};

export default Popover;
