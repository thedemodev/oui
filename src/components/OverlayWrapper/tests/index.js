import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';

import OverlayWrapper from '../index';
import Popover from '../../Popover';

const TestButton = (props) => <button>{ props.text }</button>;
const TestPopover = (props) => {
  return (
    <Popover title="Lorem ipsum dolor sit amet">
      <p>Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!</p>
    </Popover>
  );
};

TestButton.propTypes = {
  text: PropTypes.string,
};

describe('components/OverlayWrapper when componentDidMount', () => {
  it('should not instantiate tether until it is triggered for the first time', function() {
    const component = mount(
      <OverlayWrapper
        attachmentConstraint="none"
        overlay={ <TestPopover /> }
        horizontalAttachment="center"
        verticalAttachment="top"
        verticalTargetAttachment="top"
        horizontalTargetAttachment="center"
        isConstrainedToScreen={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    expect(component.instance()._tether).toBe(undefined);
  });
});

describe('components/OverlayWrapper when componentWillUnmount', () => {
  let component;
  let instance;

  beforeEach(() => {
    component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        horizontalAttachment="center"
        verticalAttachment="top"
        verticalTargetAttachment="top"
        horizontalTargetAttachment="center"
        isConstrainedToScreen={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    instance = component.instance();
    instance.enableTether();
  });

  afterEach(() => {
    component = null;
    instance = null;
  });

  it('should destroy Tether', () => {
    spyOn(instance._tether, 'destroy');

    const initialCallCount = instance._tether.destroy.calls.count();

    component.unmount();
    expect(instance._tether.destroy.calls.count()).toBe(initialCallCount + 1);
  });

  it('should call `removeBodyEventListener` function', () => {
    spyOn(instance, 'removeBodyEventListener');

    const initialCallCount = instance.removeBodyEventListener.calls.count();

    component.unmount();
    expect(instance.removeBodyEventListener.calls.count()).toBe(initialCallCount + 1);
  });
});

describe('#disableTether', () => {
  it('should call function to remove event listener from document body', () => {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();
    spyOn(instance, 'removeBodyEventListener');

    const initialCallCount = instance.removeBodyEventListener.calls.count();

    instance._tether = { disable: () => {} };
    instance.disableTether();

    expect(instance.removeBodyEventListener.calls.count()).toBe(initialCallCount + 1);
  });

  it('should call `onHide` prop if a function is provided', () => {
    const obj = { func: () => {} };
    spyOn(obj, 'func').and.stub();

    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        onHide={ obj.func }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    component.instance().enableTether();

    const initialCallCount = obj.func.calls.count();
    component.instance().disableTether();
    expect(obj.func.calls.count()).toBe(initialCallCount + 1);
  });

  it('should set visible state of `overlay` to false', () => {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    component.setState({ isOverlayOpen: true });

    const instance = component.instance();

    instance._tether = { disable: () => {} };
    instance.disableTether();

    expect(component.state('isOverlayOpen')).toBe(false);
  });

});

describe('#enableTether', () => {
  it('should only initialize tether if its not yet defined', function() {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        shouldHideOnClick={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    spyOn(component.instance(), 'initTether').and.callThrough();
    expect(component.instance().initTether.calls.count()).toBe(0);
    component.instance().enableTether();
    expect(component.instance().initTether.calls.count()).toBe(1);
    component.instance().enableTether();
    expect(component.instance().initTether.calls.count()).toBe(1);
  });

  it('should pass the correct options when all the layout props are provided', () => {
    const component = mount(
      <OverlayWrapper
        attachmentConstraint="none"
        overlay={ <TestPopover /> }
        horizontalAttachment="center"
        verticalAttachment="top"
        verticalTargetAttachment="top"
        horizontalTargetAttachment="center"
        isConstrainedToScreen={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    const instance = component.instance();
    instance.enableTether();
    const tetherOptions = instance._tether.options;
    expect(tetherOptions.attachment).toBe('top center');
    expect(tetherOptions.constraints.length).toBe(1);
    expect(tetherOptions.constraints[0]).toEqual({
      attachment: 'none',
      to: 'window',
      pin: true,
    });
    expect(tetherOptions.targetAttachment).toBe('top center');
    expect(tetherOptions.target.tagName.toLowerCase()).toBe(component.find(TestButton).render().first()[0].name);
    expect(tetherOptions.element.tagName.toLowerCase()).toBe(component.find(TestPopover).render().first()[0].name);
  });

  it('should add click event listener to document if `shouldHideOnClick` is true', () => {
    spyOn(document, 'addEventListener');

    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        shouldHideOnClick={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();

    instance._tether = { enable: () => {} };
    instance.enableTether();

    expect(document.addEventListener.calls.any()).toBe(true);
    expect(instance._bodyClickListener).toBeTruthy();
    expect(document.addEventListener.calls.argsFor(0)[0]).toBe('click');
    expect(document.addEventListener.calls.argsFor(0)[1]).toBe(instance._bodyClickListener);
  });

  it('should not add click event listener to document if `shouldHideOnClick` is false', () => {
    spyOn(document, 'addEventListener');

    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        shouldHideOnClick={ false }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();

    instance._tether = { enable: () => {} };
    instance.enableTether();

    expect(instance._bodyClickListener).toBe(false);
    expect(document.addEventListener.calls.argsFor(0)[0]).not.toBe('click');
    expect(document.addEventListener.calls.argsFor(0)[1]).not.toBe(instance._bodyClickListener);
  });

  it('should add keyup event listener to document', () => {
    spyOn(document, 'addEventListener');

    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();

    instance._tether = { enable: () => {} };
    instance.enableTether();

    expect(document.addEventListener.calls.any()).toBe(true);
    expect(document.addEventListener.calls.argsFor(1)[0]).toBe('keyup');
    expect(document.addEventListener.calls.argsFor(1)[1]).toBe(instance._documentEscapeListener);
  });

  it('should call `onShow` prop if a function is provided', () => {
    const obj = { func: () => {} };
    spyOn(obj, 'func').and.stub();

    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }
        onShow={ obj.func }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();
    const initialCallCount = obj.func.calls.count();
    instance.enableTether();
    expect(obj.func.calls.count()).toBe(initialCallCount + 1);
  });


  it('should set visible state of `overlay` to true', () => {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    component.setState({ isOverlayOpen: false });
    const instance = component.instance();
    instance._tether = { enable: () => {} };
    instance.enableTether();

    expect(component.state('isOverlayOpen')).toBe(true);
  });
});

describe('#isClickWithinOverlayOrChildren', () => {
  it('should call function to disable Tether if click is not within overlay or button', () => {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();
    spyOn(instance, 'disableTether');
    const initialCallCount = instance.disableTether.calls.count();

    instance.isClickWithinOverlayOrChildren({ target: document.body });

    expect(instance.disableTether.calls.count()).toBe(initialCallCount + 1);
  });
});

describe('#onChildClick', () => {
  let component;

  beforeEach(() => {
    component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
  });

  afterEach(() => {
    component.unmount();
    component = null;
  });

  it('should still call `children`\'s existing `onClick` if it exists', () => {
    const instance = component.instance();
    const fakeEvent = {};
    const fakeChild = { props: { onClick: () => {} } };

    spyOn(fakeChild.props, 'onClick');

    instance.onChildClick(fakeEvent, fakeChild);

    // Expect that `fakeChild`'s `onClick` was called and `fakeEvent` was
    // passed in
    expect(fakeChild.props.onClick.calls.count()).toBe(1);
    expect(fakeChild.props.onClick.calls.mostRecent().args[0]).toBe(fakeEvent);
  });

  it('should call functions to enable Tether when clicking on the children and overlay is closed', () => {
    const instance = component.instance();
    spyOn(instance, 'enableTether');
    const initialCallCount = instance.enableTether.calls.count();

    component.setState({ 'isOverlayOpen': false });

    instance.onChildClick(null, { props: {} });

    // Should call function to enable Tether
    expect(instance.enableTether.calls.count()).toBe(initialCallCount + 1);
  });

  it('should call functions to disable Tether when clicking on the children and overlay is open', () => {
    const instance = component.instance();
    spyOn(instance, 'disableTether');
    const initialCallCount = instance.disableTether.calls.count();

    component.setState({ 'isOverlayOpen': true });

    instance.onChildClick(null, { props: {} });

    // Should call function to disable Tether
    expect(instance.disableTether.calls.count()).toBe(initialCallCount + 1);
  });
});

describe('#onEscapeKey', () => {
  it('should call function to disable Tether when escape key is pressed', () => {
    const component = mount(
      <OverlayWrapper
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );

    const instance = component.instance();
    spyOn(instance, 'disableTether');
    const fakeEvent = { keyCode: 27 };

    instance.onEscapeKey(fakeEvent);

    expect(instance.disableTether.calls.count()).toBe(1);
  });
});

describe('components/OverlayWrapper snapshots & others', () => {
  it('should render a Popover component inside', () => {
    const component = mount(
      <OverlayWrapper
        behavior="click"
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render component with `state.isOverlayOpen=true` when click if click bahavior', () => {
    const component = mount(
      <OverlayWrapper
        behavior="click"
        overlay={ <TestPopover /> }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    component.find('.click-area').simulate('click');
    expect(component.state().isOverlayOpen).toBe(true);
  });
  it('should render component with `state.isOverlayOpen=true` when hover if hover behavior', () => {
    const component = mount(
      <OverlayWrapper
        behavior="hover"
        overlay={ <TestPopover /> }
        shouldHideOnClick={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    component.find('.click-area').simulate('mouseover');
    expect(component.state().isOverlayOpen).toBe(true);
  });

  it('should render component with `state.isOverlayOpen=false` when clicking escape key', () => {
    const component = mount(
      <OverlayWrapper
        behavior="hover"
        overlay={ <TestPopover /> }
        shouldHideOnClick={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    component.find('.click-area').simulate('click');
    component.simulate('keyDown', { keyCode: 40 });
    expect(component.state().isOverlayOpen).toBe(false);
  });

  it('should call prop function on click', () => {
    const component = mount(
      <OverlayWrapper
        behavior="hover"
        overlay={ <TestPopover /> }
        shouldHideOnClick={ true }>
        <TestButton text='button' />
      </OverlayWrapper>
    );
    component.find('.click-area').simulate('click');
    component.simulate('keyDown', { keyCode: 40 });
    expect(component.state().isOverlayOpen).toBe(false);
  });

});
