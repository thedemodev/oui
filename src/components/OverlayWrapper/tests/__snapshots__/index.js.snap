// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/OverlayWrapper snapshots & others should render a Popover component inside 1`] = `
<OverlayWrapper
  attachmentConstraint="together"
  behavior="click"
  horizontalAttachment="center"
  isConstrainedToScreen={false}
  onHide={[Function]}
  overlay={<TestPopover />}
  shouldHideOnClick={true}
  verticalAttachment="top"
>
  <div
    data-oui-component={true}
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <div
      className="click-area"
      onClick={[Function]}
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <TestButton
        text="button"
      >
        <button>
          button
        </button>
      </TestButton>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <TestPopover>
        <Popover
          padding="default"
          title="Lorem ipsum dolor sit amet"
        >
          <div
            className="oui-pop--over"
            data-oui-component={true}
            style={
              Object {
                "display": "block",
                "opacity": 1,
                "position": "initial",
              }
            }
          >
            <div
              className="oui-pop--over__content"
            >
              <div
                className="oui-pop--over__title"
              >
                Lorem ipsum dolor sit amet
              </div>
              <p>
                Ipsa officiis bad-news minus earum a aperiam! Aperiam reiciendis vitae nihil libero et, hic!
              </p>
            </div>
          </div>
        </Popover>
      </TestPopover>
    </div>
  </div>
</OverlayWrapper>
`;
